lib/matplotlib/collections.py

```diff
@@ -213,6 +213,22 @@ def get_window_extent(self, renderer):
         # cases other than scatter plot legend
         return self.get_datalim(transforms.IdentityTransform())
 
+    def _non_affine_transform_paths(self, paths, transform):
+        """
+        Sometimes there will be many paths, and the transform is
+        slow to run.  This helper concatenates the paths, transforms,
+        the points, and then splits them back into the individual paths.
+        """
+        vertices = np.vstack(path.vertices for path in paths)
+        pos = np.cumsum([len(x) for x in paths])
+        vertices = transform.transform_non_affine(vertices)
+        # repopulate paths
+        verts = np.split(vertices, pos[:-1])
+        for nn, verti in enumerate(verts):
+            paths[nn].vertices = verti
```

Comment:
You're modifying `paths` in place; if you trace back, this is a reference to `self._paths`, which means you're modifying the original contours. That seems really bad.